### PR TITLE
upgrade go version

### PR DESCRIPTION
**Pull request by nolouch**
ref https://github.com/tikv/pd/issues/6029 https://github.com/pingcap/tidb/issues/40969

### CODE REVIEW

**Comment by nolouch** [comment]
ptal @disksing @sticnarf @JmPotato

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.19
+        go-version: "1.20.x"
```
Comment (sticnarf) [inline .github/workflows/test.yml:18]: Can we keep this 1.19? We expect client-go itself to be compilable on the oldest officially supported Golang version.

Comment (nolouch) [reply .github/workflows/test.yml:18]: https://github.com/tikv/client-go/actions/runs/4240854947/jobs/7370351699

if we upgrade the dependency of pd or tidb, will report the error:
```
go: go.mod file indicates go 1.20, but maximum version supported by tidy is 1.19
```

Comment (nolouch) [reply .github/workflows/test.yml:18]: oh, it uses the latest TiDB. so the compatibility test will be failed always.

Comment (sticnarf) [reply .github/workflows/test.yml:18]: Ah, perhaps it's the PD client that specifies a too new toolchain...

Comment (sticnarf) [reply .github/workflows/test.yml:18]: > oh, it uses the latest TiDB. so the compatibility test will be failed always. maybe we can only change the version for compatibility test.

The compatibility tests and integration tests can use 1.20.

Comment (nolouch) [reply .github/workflows/test.yml:18]: updated in https://github.com/tikv/client-go/pull/711.

**Pull request closed without merging**